cli/src/commands/operation.rs

```diff
@@ -75,6 +81,19 @@ pub struct OperationUndoArgs {
     what: Vec<UndoWhatToRestore>,
 }
 
+/// Prune unneeded operations (EXPERIMENTAL)
```

Comment:
Yes, it's basically `op abandon`. We can even add fake opset syntax which only supports `::<op_id>` and `<op_id>::`.

I'm not sure how much we'll prefer shared/overloaded verbs. `op undo` is conceptually `op backout`. (btw, maybe we could rename `op undo` and add `undo [N]` instead to do `op restore @-...`.)

Reply:
I think it's generally better to use the same name (like with`jj restore` and `jj op restore`).

Yes, I've considered calling `jj op undo` as `jj op backout`. I wouldn't mind renaming it that way and having `jj undo` be an alias for `jj op backout`, or we can redefine `jj undo` as something else as you said.

> (btw, maybe we could rename `op undo` and add `undo [N]` instead to do `op restore @-...`.)

I'm not sure `undo [N]` would be very useful because it's not always obvious without looking at the op log how many operations there have been (and people probably don't look at the op log anywhere near as often as they look at the regular log). Another option is to add `-i` option where we let the user navigate between operations to restore to while showing them the log output at the given operation.